lib/scheduler.registry.ts

```diff
@@ -43,4 +44,23 @@ export class SchedulerRegistry {
   addTimeout(name: string, timeoutId: number) {
     this.timeouts.set(name, timeoutId);
   }
+
+  addCronJob(name: string, cronJob: CronJob) {
```

Comment:
I'm slightly afraid of having 2 methods with almost the same names (`addCron` and `addCronJob`). Maybe instead, we should add a mechanism to prevent duplicates to all of them (`addTimeout`, `addInterval`, `addCron`). Also, what if I want to dynamically create a CRON job, but start it later in a different location? I would personally remove `cronJob.start()` and leave this responsibility to a developer

Reply:
@kamilmysliwiec This update:
- uses `addCron` instead of `addCronJob`  `addCron` does not automatically start a job.  We'll document doing that.
- adds duplicate checks for `addInterval` and `addTimeout`.  However, these are probably not needed for two reasons.  I decided there's no need for a dynamic API for these (more below) and, for the static (decorator-based) API, I believe the registration happens in such a way (at compile time?) where duplicate checking does not happen.  At least not with this code.  Instead, for Intervals and Timeouts, the last declaration for a `name` overrides any previous ones.  We probably need to document this unless you don't want that behavior and know of a way to prevent it.
- adds `getIntervals` and `getTimeouts`.  Again, I added these before deciding a dynamic API is not really appropriate for intervals and timeouts.  So, these may be less useful, but I left them in for now.  We can remove them if you think they're useless.  They do return all the statically defined and registered intervals and timeouts (including unnamed ones - which is maybe weird), but since they're statically defined, the developer knows their names in the code, so....
- adds `deleteInterval` and `deleteTimeout`.  I think these are still valuable for stopping and cleaning up unused intervals and timeouts.

**No dynamic API for timeouts, intervals**

Upon further investigation, adding a dynamic API for timeouts and intervals doesn't seem to make sense.  You can't really instantiate one of these separately from registering/installing it (maybe you can, but it's definitely not standard and I've never done it, and it seemed awkward to document, unlike CronJobs), so there's really no significant benefit to creating them dynamically with our API -- the only slight benefit is registration, but in the end, if you're doing dynamic stuff you can do it all with cron jobs.